flocker/control/test/test_httpapi.py

```diff
@@ -2628,22 +2629,24 @@ def test_returns_service(self):
         """
         reactor = MemoryReactor()
         endpoint = TCP4ServerEndpoint(reactor, 6789)
-        verifyObject(IService, create_api_service(None, None, endpoint))
+        verifyObject(IService, create_api_service(
+            None, None, endpoint, ClientContextFactory()))
 
     def test_listens_endpoint(self):
         """
         ``create_api_service`` returns a service that listens using the given
-        endpoint with a HTTP server.
+        endpoint with a TLS server.
         """
         reactor = MemoryReactor()
         endpoint = TCP4ServerEndpoint(reactor, 6789)
-        service = create_api_service(None, None, endpoint)
+        service = create_api_service(None, None, endpoint,
+                                     ClientContextFactory())
         self.addCleanup(service.stopService)
         service.startService()
         server = reactor.tcpServers[0]
         port = server[0]
         factory = server[1].__class__
-        self.assertEqual((port, factory), (6789, Site))
+        self.assertEqual((port, factory), (6789, TLSMemoryBIOFactory))
```

Comment:
Not that this was a great test before ... but at least it demonstrated HTTP-ness.  It no longer does, though.